lib/spack/docs/configuration.rst

```diff
@@ -4,58 +4,136 @@
+-----------------------------------
+
+Spack's basic configuration options are set in ``config.yaml``.  You can
+see the default settings by looking at
+``etc/spack/defaults/config.yaml``:
+
+.. literalinclude:: ../../../etc/spack/defaults/config.yaml
+   :language: yaml
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Config file variables
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You may notice some variables prefixed with ``$`` in the settings above.
+Spack understands several variables that can be used in values of
+configuration parameters.  They are:
+
+  * ``$spack``: path to the prefix of this spack installation
+  * ``$tempdir``: default system temporary directory (e.g, from the
+    ``$TMPDIR`` environment variable)
+  * ``$user``: name of the current user
+
+Note that, as with shell variables, you can write these as ``$varname``
+or with braces to distinguish the variable from surrounding characters:
+``${varname}``
+
+^^^^^^^^^^^^^^^^^^^^
+``install_tree``
+^^^^^^^^^^^^^^^^^^^^
+
+Controls where Spack will install packages.  Default is
+``$spack/opt/spack``.
+
+^^^^^^^^^^^^^^^^^^^^
+``module_roots``
+^^^^^^^^^^^^^^^^^^^^
+
+Controls where Spack installs modules it generates.  You can customize
+the location for each type of module.  e.g.:
+
+.. code-block:: yaml
+
+   module_roots:
+     tcl:    $spack/share/spack/modules
+     lmod:   $spack/share/spack/lmod
+     dotkit: $spack/share/spack/dotkit
+
+^^^^^^^^^^^^^^^^^^^
+``build_stage``
+^^^^^^^^^^^^^^^^^^^
+
+Spack is designed to run out of a user home directories, and on many
+systems the home directory is NFS mounted *not* a fast filesystem.  We
+create build stages in a temporary directory to avoid this.  Many systems
+impose quotas on home directories, and ``/tmp`` or similar directories
+often have more available space.
+
+By default, the ``build_stage`` is set like this:
+
+.. code-block:: yaml
+
+   build_stage:
+    - $tempdir
+    - /nfs/tmp2/$user
+    - $spack/var/spack/stage
+
+This is an ordered list of paths that Spack should search when trying to
+find a temporary directory for the build stage.  The list is searched in
+order, and Spack will use the first directory to which it has write access.
+
+When Spack creates a temporary build stage, it places a symbolic link to
+the stage in ``$spack/var/spack/stage`` so that it can track the stage
+and clean it up later.  By default, build stages are removed after a
+successful package build.  You can manually clean build stages with
+:ref:`spack purge <cmd-spack-purge>`.
+
+Note that the last item in the list is ``$spack/var/spack/stage``.  If
+this is used as the build stage, Spack will build *directly* in its
+prefix and will not use temporary space.
+
+^^^^^^^^^^^^^^^^^^^^
+``source_cache``
+^^^^^^^^^^^^^^^^^^^^
+
+Location to cache downloaded tarballs and repositories.  By default these
+are stored in ``$spack/var/spack/cache``.  These are stored indefinitely
+by default. See :ref:`spack purge <cmd-spack-purge>`.
+
```

Comment:
The config file explains how this can be purged; should probably also be in the rst docs.